python/ray/dag/compiled_dag_node.py

```diff
@@ -1111,6 +1125,14 @@ def _add_edge(
                 # Add an edge from the writer to the reader.
                 _add_edge(graph, idx, downstream_idx)
                 if task.dag_node.type_hint.requires_nccl():
+                    if _is_same_actor(idx, downstream_idx):
```

Comment:
What does it exactly mean? it means task A and task B runs on the same actor?

Reply:
>  it means task A and task B runs on the same actor?

Yes. NCCL is a blocking operation, which means the writer and the reader need to write and read at the same time. However, an actor can only execute one Ray task at a time. Hence, it is impossible for both the reader and writer to operate simultaneously, so the NCCL operation will block forever.